provisioning/image_cache.go

```diff
@@ -138,11 +167,17 @@ func newImageCacheContainers(images *Images, proxy *osconfigv1.Proxy) []corev1.C
 }
 
 func newImageCachePodTemplateSpec(info *ProvisioningInfo) (*corev1.PodTemplateSpec, error) {
-	cacheConfig, err := imageCacheConfig(info.Namespace, info.ProvConfig.Spec)
+	imageURL, err := imageCacheConfig(info.Namespace, info.ProvConfig.Spec)
 	if err != nil {
 		return nil, err
 	}
-	info.ProvConfig.Spec = *cacheConfig
+
+	initContainers := []corev1.Container{
+		createInitContainerMachineOsDownloader(info, *imageURL, false),
```

Comment:
The same logic is required as in the metal3 Deployment - we may need two MachineOsDownloader containers.

Reply:
I think adding the parameter fixes this issue. Correct?

Reply:
If it's not called twice then it isn't fixed.
Also the parameter hasn't been added yet.

Reply:
I overhauled the image cache code a bit to fix this. PTAL. Thanks